var/spack/repos/builtin/packages/timemory/package.py

```diff
@@ -75,7 +101,19 @@ def cmake_args(self):
         else:
             args.append('-DTIMEMORY_USE_MPI=OFF')
 
+        if '+gotcha' in spec:
+            args.append('-DTIMEMORY_USE_GOTCHA=ON')
+        else:
+            args.append('-DTIMEMORY_USE_GOTCHA=OFF')
+
+        if '+cuda' in spec:
+            args.append('-DTIMEMORY_USE_CUDA=ON')
+        elif '+cupti' not in spec:
+            # CUPTI requires CUDA
```

Comment:
You can add the following line below the dependencies listed above:
```python
conflicts('+cupti', when='~cuda', msg='CUPTI requires CUDA')
```
That way, it will be impossible to build `+cupti~cuda`.